### PR TITLE
[REVIEW] PR addressing the implementation of series.where()

### LINKED ISSUE #1514: [FEA] Implement Series.where
**Is your feature request related to a problem? Please describe.**
I'd like to be able to use `Series.where(...)` in the same way I would do in pandas. This partially blocks dask `cummin` and `cummax` operations (but not sum and product). The API docs are [here](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.Series.where.html). `where` is essentially an API for replacing values based on a supplied condition that evaluates to a Boolean array.

We can include an assert that requires the `axis` argument to be zero, as we will need that keyword argument downstream regardless of whether it's implemented. `Where` has a lot of complex functionality that may not be necessary to implement all at once. For dask cummin/cummax, minimally sufficient functionality would include the ability to replace all values with a scalar `other` where `cond` evaluates to False.

```python
s = pd.Series(range(5))
s.where(s > 1, 10)
0    10
1    10
2    2
3    3
4    4
dtype: int64
```

### PR DESCRIPTION
**Pull request by AK-ayush**
Once done, this PR closes #1514 . currently depends on #1560.

### CODE REVIEW

File: python/cudf/dataframe/series.py [hunk 1]
```diff
@@ -689,6 +689,39 @@ def fillna(self, value, method=None, axis=None, inplace=False, limit=None):
...
+            replace with corresponding value from other.
+        :param other: scalar
+            Entries where cond is False are replaced with
+            corresponding value from other.
+        :param axis:
+        :return: Series
+
+        """
+        idx_bool = cond.to_array()
+        to_replace = self.to_array()[idx_bool]
```
Comment (kkraus14) [inline python/cudf/dataframe/series.py]: Should keep both of these on the GPU instead of converting to numpy arrays in host memory.

Comment (kkraus14) [inline python/cudf/dataframe/series.py]: These won't handle nulls, so suggest you assert the null count is zero.

Comment (AK-ayush) [reply python/cudf/dataframe/series.py]: That was a kind of draft. Please ignore them.

**Comment by AK-ayush** [comment]
I have added the changes required for this feature on the basis of new implementation of `Series.replace()`, which is yet to be merged.

**Pull request merged**